### PR TITLE
Enable editor step tests

### PR DESCRIPTION
**Pull request by sangeetha-nayak**
## Proposed changes

[PRESS0-1823](https://jira.newfold.com/browse/PRESS0-1823) | Enable editor step tests for AI Sitegen

## Type of Change

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/bluehost/.github/blob/master/.github/contributing.md) doc
- [x] I have viewed my change in a web-browser
- [x] Linting and tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: tests/cypress/integration/5-AI-SiteGen-onboarding-flow/7-editor.cy.js [hunk 1]
```diff
@@ -103,7 +104,7 @@ describe.skip( 'SiteGen Site Editor Step', function () {
 			'.nfd-onboarding-sidebar--sitegen-editor-patterns__header__tab-panel__versions-tab__preview-container'
 		)
 			.should( 'be.visible' )
-			.should( 'have.length', 3 ); // as we are not regenrating new themes
+			.should( 'have.length', 4 ); // Regenrating new themes
```
Comment (officiallygod) [inline tests/cypress/integration/5-AI-SiteGen-onboarding-flow/7-editor.cy.js:107]:
```suggestion
			.should( 'have.length', 4 ); // Regenerating new themes
```

File: tests/cypress/integration/5-AI-SiteGen-onboarding-flow/6-preview.cy.js
Comment (officiallygod) [inline tests/cypress/integration/5-AI-SiteGen-onboarding-flow/6-preview.cy.js:1]: Should we not also add a test case where we ensure Back button is not visible as it was removed first and later found out it still exists in a bug and fixed, Maybe we can add a test to ensure no one introduces that again?

Same for all steps that should not have a back button.

File: tests/cypress/integration/5-AI-SiteGen-onboarding-flow/4-site-logo.cy.js [hunk 2]
```diff
@@ -96,4 +96,11 @@ describe( 'SiteGen Site Logo Step', function () {
 			.click();
 		cy.url().should( 'not.contain', 'sitegen/step/site-logo' );
 	} );
+	
+	it('Verify Skip for now and go to next step', () => {
+		cy.get( '.nfd-onboarding-step--site-gen__site-logo__container__buttons__skip' )
+			.should( 'not.be.disabled' )
+			.click();
+		cy.url().should( 'not.contain', 'sitegen/step/site-logo' );
+	} )
```
[the remark this answers is not in the document]
Comment (sangeetha-nayak) [reply tests/cypress/integration/5-AI-SiteGen-onboarding-flow/4-site-logo.cy.js]: Valid. Updated it.

File: tests/cypress/integration/5-AI-SiteGen-onboarding-flow/6-preview.cy.js [hunk 3]
```diff
@@ -106,7 +106,7 @@ describe( 'SiteGen Site Preview Step', function () {
 			.contains( 'Favorite' );
```
[the remark this answers is not in the document]
Comment (sangeetha-nayak) [reply tests/cypress/integration/5-AI-SiteGen-onboarding-flow/6-preview.cy.js:110]: This has to be done throughout General, ecommerce and AI onboarding. Will have to pick this up as a separate story.

File: tests/cypress/integration/5-AI-SiteGen-onboarding-flow/6-preview.cy.js
Comment (sangeetha-nayak) [reply tests/cypress/integration/5-AI-SiteGen-onboarding-flow/6-preview.cy.js:1]: Added in experience, preview and editor. Hope that is all.

File: tests/cypress/integration/5-AI-SiteGen-onboarding-flow/6-preview.cy.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (officiallygod) [reply tests/cypress/integration/5-AI-SiteGen-onboarding-flow/6-preview.cy.js:110]: Sure, Please make sure nowhere do we use hardcoded string as a point of truth, Thanks.

File: tests/cypress/integration/5-AI-SiteGen-onboarding-flow/7-editor.cy.js [hunk 4]
```diff
@@ -282,4 +281,11 @@ describe.skip( 'SiteGen Site Editor Step', function () {
 			'.nfd-onboarding-sidebar--customize__design-colors-panel__custom__colors__container'
 		).should( 'not.exist' );
 	} );
+	
+	it( 'Check Save & Continue button functionality', () => {
+		cy.get('.nfd-onboarding-header--sitegen__editor__end__save-button__text', { timeout : 20000 } )
+			.click();
+		cy.get(`.${getAppId()}-app-container`, { timeout : 90000 } )
```
Comment (officiallygod) [inline tests/cypress/integration/5-AI-SiteGen-onboarding-flow/7-editor.cy.js:288]: Last doubt are the files linted just curious?

File: tests/cypress/integration/5-AI-SiteGen-onboarding-flow/6-preview.cy.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (sangeetha-nayak) [reply tests/cypress/integration/5-AI-SiteGen-onboarding-flow/6-preview.cy.js:110]: Please create a task for this in backlog, someone among QA should be able to slowly start on this.

File: tests/cypress/integration/5-AI-SiteGen-onboarding-flow/7-editor.cy.js (hunk 4 above)
Comment (sangeetha-nayak) [reply tests/cypress/integration/5-AI-SiteGen-onboarding-flow/7-editor.cy.js]: Done

**Review by officiallygod — APPROVED** [review]

**Pull request merged**